### DISCUSSION TITLE
MITgcm periodic boundary

### DISCUSSION THREAD

**Discussion by ah-dinh** [Parcels-code/Parcels#1584] [Q&A]
Hi, 

I am trying to use periodic (longitude) boundary on an MITgcm dataset.

XG[0] = -180 
XG[-1] = 179.95834 

I get a "FieldOutOfBoundError" when I try to sample the last column of the dataset, basically anything from 179.95834  to 180. 

I see in the tutorial that a halo and a periodic boundary kernel is need for an A-grid dataset. MITgcm is C-grid, but do I need to do something similar as well to get this to work?

![Screenshot 2024-06-03 at 13 53 22](https://github.com/OceanParcels/parcels/assets/69325452/93de8638-fde7-4d35-aea8-749841400427)

**Comment by erikvansebille** [comment]
Hi @ah-dinh, thanks for asking. While normally we only advice to use `fieldset.add_periodic_halo()` for A-grid, there is no a-priori reason it wouldn't work on C-grids. 

Perhaps the confusion lies in the point that we advise against using halo's for ORCA (NEMO) grids, because these already include a halo (i.e. `XG[-1] > XG[0]` in your notation above). 

If that is not the case in your MITgcm grid, then using a halo would be a good fix I reckon. Alternatively, you can also try the 'hycom-solution' (adding an extra column of longitudes at the end of the longitude array) proposed in the second cell of [the periodic boundaries tutorial](https://docs.oceanparcels.org/en/latest/examples/tutorial_periodic_boundaries.html).
